### ISSUE TITLE
switch to mocha for new tests? 

### ISSUE THREAD

**Issue by kevinburkeshyp** [coopernurse/node-pool#126]
I'm currently trying to write a test, and observing that adding the following snippet of code to one test:

```javascript
      setTimeout(function() {
        pool.release(obj1)
      }, 2)
```

causes 9 other tests to fail. I'm not sure what or why this is happening, but given the test runner (expresso) is deprecated, maybe we could switch to mocha for new tests ?

I'm also curious why that's happening and would like to investigate/fix, but I'm not sure the benefit there outweighs the time investment.

**Comment by kevinburkeshyp** [comment]
I guess the `setTimeout` causes the test to throw an assertion error, and because all of the tests are running in parallel, this isn't handled anywhere, and causes other tests to also fail. OK.

**Comment by sandfox** [comment]
yeah, there is an outstanding issue for this #75 but I've not had any time to work on it. my personal preference is for tap/tape but I'm open to just replacing the tests with mocha if someone else finds it easier. The logic/structure of the tests could really do with updating too so it's kind of a 2 phase problem - making the tests make sense to the viewer, and converting to mocha/tap.

**Comment by sandfox** [comment]
Closing as duplicate of #75, can carry on there
